src/nyc_trees/apps/quiz/models.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+
+
+class Quiz(object):
+    """
+    title - string
+    questions - iterable<Question>
+    passing_score - int; How many correct answers are needed to pass the quiz
+    """
+    def __init__(self, title, questions, passing_score):
+        assert passing_score <= len(questions)
+        self.title = unicode(title)
+        self.questions = list(questions)
+        self.passing_score = int(passing_score)
+
+    @classmethod
+    def extract_answers(cls, post_fields):
+        """Return dict of {answer_index => selected_choice_index, ...}
+        post_fields - QueryDict; Probably from request.POST
+        """
+        result = []
+        for field_name, values in post_fields.iteritems():
+            # Parse field name format like "question.1", "question.2", etc.
+            parts = field_name.split('.')
+            if len(parts) == 2:
+                try:
+                    question_index = int(parts[1], 10)
+                    value = int(values[0])
+                    result.append((question_index, value))
+                except ValueError:
+                    pass
+        return dict(result)
+
+    def score(self, answers):
+        return sum(1 for i, question in enumerate(self.questions)
```

Comment:
fwiw, this "bare generator" syntax has a tendency to confuse developers. Consider putting a set of `[]` to make it an intermediate list via comprehension, since the size of `self.questions` will always be small.

more context:
http://python-history.blogspot.com/2010/06/from-list-comprehensions-to-generator.html (search for 'syntax')